### PR TITLE
MULE-18895: Avoid cursor re-management with PayloadStatistics decorators

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

**Comment by elrodro83** [comment]
--validate

**Comment by elrodro83** [comment]
--validate

**Comment by elrodro83** [comment]
--validate

**Comment by elrodro83** [comment]
--validate

**Comment by elrodro83** [comment]
--validate

**Comment by elrodro83** [comment]
--validate

**Comment by elrodro83** [comment]
--validate

File: core-tests/src/test/java/org/mule/runtime/core/internal/management/stats/PayloadStatisticsTestCase.java [hunk 1]
```diff
@@ -766,6 +772,45 @@ public void flowProcessMediatorCollectionItems() throws Exception {
...
+        new InputDecoratedCursorStreamProvider(provider, decoratorFactory.componentDecoratorFactory(component1), CORR_ID);
+
+    final CursorProvider managed = streamingManager.manage(decoratedProvider, testEvent());
+
+    assertThat(managed, is(sameInstance(decoratedProvider)));
+  }
+
+  @Test
+  @Issue("MULE-18895")
+  @Description("Check that decorated cursors also are cursors so the fucntionality depending on instanceof is not affected.")
```
Comment (fsgonz) [inline core-tests/src/test/java/org/mule/runtime/core/internal/management/stats/PayloadStatisticsTestCase.java:794]: functionality

**Review by fsgonz — CHANGES_REQUESTED** [review]

File: core-tests/src/test/java/org/mule/runtime/core/internal/management/stats/PayloadStatisticsTestCase.java (hunk 1 above)
Comment (fsgonz) [reply core-tests/src/test/java/org/mule/runtime/core/internal/management/stats/PayloadStatisticsTestCase.java:794]: so that

**Review by fsgonz — APPROVED** [review]

**Pull request merged**